build/scripts/version-update/main.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"flag"
+	"io/ioutil"
+	"log"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var releaseStage string
+
+func init() {
+	flag.StringVar(&releaseStage, "release-stage", "", "Specify the release stage ('before' or 'after')")
+}
+
+func main() {
+	flag.Parse()
+
+	if len(flag.Args()) < 1 {
+		log.Fatalf("Please provide the release stage ('before' or 'after') and the initial version as command-line arguments")
+	}
+
+	initialVersion := flag.Arg(0)
+
+	log.Printf("Initial Version: %s", initialVersion)
+
+	files := []string{
+		"install/helm/agones/Chart.yaml",
+		"install/yaml/install.yaml",
+		"install/helm/agones/values.yaml",
+		"sdks/nodejs/package.json",
+		"sdks/nodejs/package-lock.json",
+		"sdks/unity/package.json",
+		"sdks/csharp/sdk/AgonesSDK.nuspec",
+		"sdks/csharp/sdk/csharp-sdk.csproj",
+	}
+
+	for _, filename := range files {
+		err := UpdateFile(filename, initialVersion)
+		if err != nil {
+			log.Fatalf("Error updating file %s: %s\n", filename, err.Error())
+		}
+	}
+}
+
+func UpdateFile(filename string, initialVersion string) error {
+	fileBytes, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+
+	content := string(fileBytes)
+
+	ext := filepath.Ext(filename)
+	validExtensions := []string{".yaml", ".yml", ".nuspec", ".csproj"}
```

Comment:
I would eliminate this and just change the below to `if ext == ".json"` - you're adding some amount of complexity just to validate a list that is more or less a constant, 20 lines previous to this.

Eliminating this will allow you to eliminate the logic in lines 62-72 and 86-90 as well - it seems like we're going through a lot of effort there.

Reply:
Updated the code and it looks simple now. Thanks Zach!